src/platform/tasks/vscode/tasksService.ts

```diff
@@ -42,6 +57,44 @@ export class TasksService implements ITasksService {
 		return existingTasks.some(t => this.matchesTask(t, def));
 	}
 
+	/**
+	 * This is needed because when tasks exit, they're removed from the taskExecutions, but we might want to review the output of the task
+	 * after it has exited. This allows us to get the terminal for a task definition.
+	 * @param task
+	 */
+	getTerminalForTask(taskDefinition: vscode.TaskDefinition): vscode.Terminal | undefined {
+		for (const [key, terminal] of this.taskDefinitionToTerminal.entries()) {
+			if (key.id) {
+				// Only some task definitions have IDs
+				const taskId = this.getTaskId(key)
+				if (taskId === key.id) {
+					return;
+				}
+			}
+			if ((taskDefinition.type === key.type &&
+				(!key.label || taskDefinition.label === key.label) &&
+				(!key.script || taskDefinition.script === key.script) &&
+				(!key.command || taskDefinition.command === key.command))) {
+				return terminal;
+			}
+		}
+	}
+
+	getTaskId(taskDefinition: vscode.TaskDefinition): string | undefined {
```

Comment:
this comes from here https://github.com/microsoft/vscode/blob/0748e889c18201cf0040c9022e5f87d4aaabf249/src/vs/workbench/contrib/tasks/common/tasks.ts#L1296-L1317